runtime/include/QuantumDevice.hpp

```diff
@@ -198,71 +198,71 @@ struct QuantumDevice {
     /**
      * @brief Get the state-vector of a device.
      *
-     * @param state The pre-allocated `MemRefView<complex<double>, 1>`
+     * @param state The pre-allocated `DataView<complex<double>, 1>`
      */
-    virtual void State(MemRefView<std::complex<double>, 1> &state) = 0;
+    virtual void State(DataView<std::complex<double>, 1> &state) = 0;
 
     /**
      * @brief Compute the probabilities of each computational basis state.
 
-     * @param probs The pre-allocated `MemRefView<double, 1>`
+     * @param probs The pre-allocated `DataView<double, 1>`
      */
-    virtual void Probs(MemRefView<double, 1> &probs) = 0;
+    virtual void Probs(DataView<double, 1> &probs) = 0;
 
     /**
      * @brief Compute the probabilities for a subset of the full system.
      *
-     * @param probs The pre-allocated `MemRefView<double, 1>`
+     * @param probs The pre-allocated `DataView<double, 1>`
      * @param wires Wires will restrict probabilities to a subset of the full system
      */
-    virtual void PartialProbs(MemRefView<double, 1> &probs,
+    virtual void PartialProbs(DataView<double, 1> &probs,
                               const std::vector<QubitIdType> &wires) = 0;
 
     /**
      * @brief Compute samples with the number of shots on the entire wires,
      * returing raw samples.
      *
-     * @param samples The pre-allocated `MemRefView<double, 2>`representing a matrix of
-     * shape `shots * numQubits`. The built-in iterator in `MemRefView<double, 2>`
+     * @param samples The pre-allocated `DataView<double, 2>`representing a matrix of
+     * shape `shots * numQubits`. The built-in iterator in `DataView<double, 2>`
      * iterates over all elements of `samples` row-wise.
      * @param shots The number of shots
      */
-    virtual void Sample(MemRefView<double, 2> &samples, size_t shots) = 0;
+    virtual void Sample(DataView<double, 2> &samples, size_t shots) = 0;
 
     /**
      * @brief Compute partial samples with the number of shots on `wires`,
      * returing raw samples.
      *
-     * @param samples The pre-allocated `MemRefView<double, 2>`representing a matrix of
-     * shape `shots * numWires`. The built-in iterator in `MemRefView<double, 2>`
+     * @param samples The pre-allocated `DataView<double, 2>`representing a matrix of
+     * shape `shots * numWires`. The built-in iterator in `DataView<double, 2>`
      * iterates over all elements of `samples` row-wise.
```

Comment:
What does row-wise mean (I can never get that straight)? 😅 Do we iterate over shots or numWires first?

Reply:
We iterator over wires for each shot (row) that's row-wise traversing through the matrix.